### PR TITLE
refactor visualization

### PR DESCRIPTION
**Pull request by JakobBD**
Just FYI @SallyDa and @Merjo that I'm working on this. 

No need to review yet. 

Haven't tested yet, haven't adapted examples and simson yet. 

@SallyDa as you can see, when I remove the Sankey plotting from the DataWriter, the only remaining class attribute is the MFASystem, which seems like a hint to move it into that class... though I like having it in a separate file. What do you think?

### CODE REVIEW

**Comment by SallyDa** [comment]
I'd be inclined to keep it separate. I think this is the kind of thing (like DataReader) that people might want to make their own functionality for, depending on exactly what they would like to save and where. It is then easy to imagine having multiple different MFASystems (multiple user defined child classes) with some standardised output, i.e. the same DataReader class.

**Comment by JakobBD** [comment]
> I'd be inclined to keep it separate

Cool! turns out I can't pass the `mfa` object in simson, as it's not yet initialized, so there is really no reason to have the functions in a class at all, so I made them functions, but kept them separate.

**Comment by SallyDa** [comment]
I'm not completely convinced by your design decisions, neither that ArrayPlotter should inherit from pydantic.BaseModel, nor that the data writing functions don't belong in a class, but I don't really have arguments for doing it differently either. We can leave it as you have it for now, and maybe revisit later if we add / update functionality.

I didn't go deep into the details of your changes, but I did look at the usecase in example 2, which seems fine. My only small comments there are:
* you are losing the plot titles when you have multiple plots on the same fig_ax
* it seems strange that you are producing two plots for Mn, Cu when they are already in the same MFASystem, and then putting them on one fig_ax, rather than selecting this as a dimension. The opposite case, where you select from the dimension is illustrated in the final plot, but for this, you have to do some data manipulation first to put it all into one array. I would rather demonstrate the multi-line plots with the first Mn, Cu cases, and then use multiple PlotlyArrayPlotter instances for the "scenario" case at the end.

File: sodym/export/data_writer.py [hunk 1]
```diff
@@ -1,139 +1,64 @@
...
-        if not os.path.exists(export_directory):
-            os.makedirs(export_directory)
-        for stock_name, stock in mfa.stocks.items():
-            df = stock.stock.to_df()
-            path_out = os.path.join(export_directory, f'{to_valid_file_name(stock_name)}_stock.csv')
+def export_mfa_to_pickle(mfa, export_path: str):
+    dict_out = convert_to_dict(mfa)
+    pickle.dump(dict_out, open(export_path, "wb"))
+    logging.info(f'Data saved to {export_path}')
+
```
Comment (SallyDa) [inline sodym/export/data_writer.py:15]: Very picky pep-y comment: top level functions (not class methods), should be surrounded by two blank lines.

**Comment by JakobBD** [comment]
addressed all points

**Comment by JakobBD** [comment]
Thanks for the review, those were all good points!

**Pull request merged**